src/main/kotlin/com/safehill/kclient/tasks/BackgroundOperationProcessor.kt

```diff
@@ -0,0 +1,75 @@
+package com.safehill.kclient.tasks
+
+import java.util.*
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+
+open class BackgroundOperationProcessor<T : AbstractBackgroundOperation>(
+    delayedStartInSeconds: Int,
+    dispatchIntervalInSeconds: Int?
+) {
+    private val dispatchIntervalInSeconds: Int?
+    private val delayedStartInSeconds: Int
+    private var started = false
+    private val stateLock = Any()
+    private val timerLock = Any()
+    private var timer: Timer? = null
+    private val timerExecutor = Executors.newScheduledThreadPool(1)
+    // TODO: Review thread pool size
+    private val operationQueue: ExecutorService = Executors.newFixedThreadPool(10);
+
+    init {
+        require(delayedStartInSeconds >= 0) { "Delayed start must be non-negative." }
+        if (dispatchIntervalInSeconds != null && dispatchIntervalInSeconds < 0) {
+            throw IllegalArgumentException("Dispatch interval must be non-negative.")
+        }
+        this.delayedStartInSeconds = delayedStartInSeconds
+        this.dispatchIntervalInSeconds = dispatchIntervalInSeconds
+    }
+
+    fun repeat(operation: T) {
+        synchronized(stateLock) {
+            if (started) return
+            started = true
+        }
+        scheduleTask(operation, delayedStartInSeconds)
+    }
+
+    private fun scheduleTask(operation: T, seconds: Int) {
```

Comment:
Can we consider using Coroutines (kotlinx-coroutines) as a lightweight and kotlin-friendly alternative to scheduled tasks?
https://kotlinlang.org/docs/coroutines-basics.html#your-first-coroutine